common/debug-interface/src/lib.rs

```diff
@@ -59,3 +59,56 @@ impl NodeDebugClient {
         Ok(response.json()?)
     }
 }
+
+/// Implement default utility client for AsyncNodeDebugInterface
+pub struct AsyncNodeDebugClient {
+    client: reqwest::Client,
+    addr: String,
+}
+
+impl AsyncNodeDebugClient {
+    /// Create AsyncNodeDebugInterface from a valid socket address.
+    pub fn new<A: AsRef<str>>(address: A, port: u16) -> Self {
```

Comment:
Can we pass reqwest::Client here? It is a good practice to reuse same Client to save resources

Reply:
I don't quite get it. Where is the same Client from? Like we check if there is any client for this IP address first?

Reply:
We just want to one Client instance and pass this (cloned) instance into each  AsyncNodeDebugClient , as an example you can look at how `Cluster::from_host_port` shares single Client across all Instance.

[As a note, cloning Client only clones 'handle', but resources are shared]

You don't need to check one client per IP or smth like that.

Reply:
Can we reuse the http Client in Cluster for this purpose? I don't know what is the http client used for?

Reply:
Yes, we ca reuse http client from cluster, or create a new one, just don't want to create O(N) http clients